app/vue/contexts/competition/SectionSchedulesContext.js

```diff
@@ -87,14 +87,62 @@ export default class SectionSchedulesContext extends BaseAppContext {
   }
 
   /**
-   * Format late registration end date.
+   * Generate late registration end label.
    *
    * @returns {string}
    */
-  formatLateRegistrationEndDate () {
-    return this.normalizeTimestamp({
+  generateLateRegistrationEndLabel () {
+    if (this.hasRegistrationPeriodEnded()) {
+      return 'Registration period has ended on'
+    }
+
+    return 'Registration period will end on'
+  }
+
+  /**
+   * Generate late registration end date timestamp.
+   *
+   * @returns {string}
+   */
+  generateLateRegistrationEndTimestamp () {
+    const normalizedTimestamp = this.normalizeTimestamp({
       timestamp: this.extractLateRegistrationEndDateString(),
     })
+
+    if (this.hasRegistrationPeriodEnded()) {
+      return `${normalizedTimestamp} ⏱️`
+    }
+
+    return normalizedTimestamp
+  }
+
+  /**
+   * Check if the registration period has ended.
+   *
+   * @returns {boolean}
+   */
+  hasRegistrationPeriodEnded () {
+    const dateString = this.extractLateRegistrationEndDateString()
+
+    if (!dateString) {
+      return true
+    }
+
+    const now = new Date()
```

Comment:
@hecticme 

In future, this style is important for test.
Because it makes test to use mock datetime easily.

```suggestion
    const now = this.createCurrentDatetime()
```